rskj-core/build.gradle

```diff
@@ -1,10 +1,127 @@
 plugins {
     id 'application'
+    id 'checkstyle'
+    id "com.diffplug.spotless" version "6.13.0"
 }
 
 apply plugin: 'maven-publish'
 apply plugin: 'jacoco'
 
+import java.nio.file.Paths
+
+// Define static timestamp
+def staticTimestamp = '2024-09-03T02:17:00'
```

Comment:
issue:

If this is fixed to a current date, does it mean that if we rebase or merge after this date but a file that wasn't supposed to be checked with a date after this one could be included in our commit? Because we have a lot of feature branches that lives for a long time (like the snap synch one) and we keep merging master to it to update. With this logic, will these files from master be included or not in the checking?

Maybe worth to test this on the snap-synch branch? 

A possibility could be passed the cutoff date as parameter... 🤔

Reply:
Thanks for bringing this up! I see where you're coming from regarding the fixed timestamp. The idea was to enforce code conventions from a specific point in time, so any code merged or rebased after that date would follow the latest standards. The assumption was that branches or PRs that are older would need to rebase or merge from master to align with these conventions

However, I understand this might not work perfectly for long-lived branches that are regularly updated from master

What are your thoughts on this? One idea could be to keep the timestamp in a file and update it manually as needed, or perhaps use a GitHub Actions workflow to manage it

Let me know if you have other suggestions or ideas!